Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -2253,6 +2253,8 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
             // FIXME: Is there any case not involving NaN where x === x is not guaranteed to return true?
             // If not I might slightly simplify that check.
             if (node->isBinaryUseKind(BooleanUse)
+                || node->isBinaryUseKind(BooleanUse, UntypedUse)
+                || node->isBinaryUseKind(UntypedUse, BooleanUse)
```

Comment:
Hmm.. How is that possible the node has two same nodes but with different used kinds?

Reply:
See FixupPhase. We have a case, which only do speculation on the one part of edge.